### PR TITLE
Adding --build B, --stage, --extract E flags

### PR DESCRIPTION
**Pull request by fejta**
ref #1475 

`--build` will continue to build kubernetes same as at present
`--build=B` will build using the `B` strategy, where `B` is `bazel` `quick` or `release`

`--stage=gs://foo/bar` will call `kubernetes/release/push-build.sh --bucket=gs://foo/bar`

`--extract=E` will download and call https://get.k8s.io with the appropriate `KUBERNETES_*` vars set. `E` can be a variety of values `v1.5.0-beta.2`, `release/stable`, `ci/latest-1.5`, `gke`, `gci/FAMILY` etc

Corresponding logic in `e2e-runner.sh` is converted to setting the appropriate `e2e.go` flags

### CODE REVIEW

**Comment by fejta** [comment]
Technically the --build and --stage flags are not yet necessary (they will only be used by the pull jobs). 

Would you prefer to review all three flags or would you like me to limit this to just the --extract flag?

**Comment by spxtr** [comment]
Either way is fine with me, but it might be hard to know what the other two flags are supposed to do if they're not used anywhere.

**Comment by fejta** [comment]
Once this is in I'm going to make pull-kubernetes-e2e, etc do everything inside the container and use it

**Comment by fejta** [comment]
Canary results for an image that uses this: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-canary/100

**Comment by krzyzacy** [comment]
Document this somewhere?

**Comment by krzyzacy** [comment]
canary is unhappy :-(

File: kubetest/main.go [hunk 1]
```diff
@@ -112,7 +216,256 @@ func main() {
...
+}
+
+func postpare(publish, save, url, version string) error {
+	if publish != "" {
+		if v, err := ioutil.ReadFile("version"); err != nil {
+			return err
+		} else {
+			log.Printf("Set %s version to %s", publish, string(v))
+		}
+		if err := finishRunning(exec.Command("gcloud", "cp", "version", publish)); err != nil {
```
Comment (krzyzacy) [inline kubetest/main.go]: gsutil cp

File: kubetest/main.go [hunk 2]
```diff
@@ -112,7 +216,256 @@ func main() {
...
+			return err
+		}
+	}
+	if save != "" {
+		if err := finishRunning(exec.Command("gcloud", "cp", home(".kube", "config"), filepath.Join(save, "kube-config"))); err != nil {
```
Comment (krzyzacy) [inline kubetest/main.go]: and here

File: kubetest/main.go (hunk 1 above)
Comment (fejta) [inline kubetest/main.go]: Done

File: kubetest/main.go (hunk 2 above)
Comment (fejta) [inline kubetest/main.go]: Done

**Comment by fejta** [comment]
New results will be at https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-canary/102

**Comment by krzyzacy** [comment]
/lgtm

update the tag?

**Comment by fejta** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-canary/107

**Pull request merged**